File: Baekjoon/Stack/1874.py
# https://www.acmicpc.net/problem/1874

n = int(input())
stack = []
answer = []
find = True
now = 1

for _ in range(n):
    num = int(input())
    
    while now <= num:
        stack.append(now)
        answer.append('+')
        now += 1
        
    if stack[-1] == num:
        stack.pop()
        answer.append('-')

    else:
        find = False

if not find:
    print('NO')
else:
    for i in answer :
        print(i)

File: Baekjoon/Stack/1874.cpp
// https://www.acmicpc.net/problem/1874

#include <iostream>
#include <vector>
#include <stack>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    stack<int> s;
    vector<char> result;
    int i = 0;
    for (int x = 1; x <= n; x++) {
        s.push(x);
        result.push_back('+');
        while (!s.empty() && s.top() == a[i]) {
            s.pop();
            result.push_back('-');
            i++;
        }
    }
    if (!s.empty()) {
        cout << "NO\n";
    } else {
        for (char c : result) {
            cout << c << '\n';
        }
    }
    return 0;
}